src/tabpfn_extensions/finetune/finetune_classifier.py

```diff
@@ -0,0 +1,395 @@
+    )
+    eval_classifier.fit(X_train, y_train)
+
+    try:
+        probabilities = eval_classifier.predict_proba(X_val)  # type: ignore
+        roc_auc = (
+            roc_auc_score(
+                y_val,
+                probabilities,
+                multi_class="ovr",
+                average="weighted",
+            )
+            if len(np.unique(y_val)) > 2
+            else roc_auc_score(
+                y_val,
+                probabilities[:, 1],
+            )
+        )
+        log_loss_score = log_loss(y_val, probabilities)
+    except (ValueError, RuntimeError, AttributeError) as e:
+        logging.warning(f"An error occurred during evaluation: {e}")
+        roc_auc, log_loss_score = np.nan, np.nan
+
+    return roc_auc, log_loss_score  # pyright: ignore[reportReturnType]
+
+
+class FinetunedTabPFNClassifier(BaseEstimator, ClassifierMixin):
+    """A scikit-learn compatible wrapper for fine-tuning the TabPFNClassifier.
+
+    This class encapsulates the fine-tuning loop, allowing you to fine-tune
+    TabPFN on a specific dataset using the familiar .fit() and .predict() API.
+
+    Parameters:
+    ----------
+    device : str, default='cuda'
+        The device to run the model on, e.g., 'cuda' or 'cpu'.
+
+    epochs : int, default=5
+        The total number of passes through the fine-tuning data.
+
+    learning_rate : float, default=1e-5
+        The learning rate for the Adam optimizer. A small value is crucial
+        for stable fine-tuning.
+
+    n_inference_context_samples : int, default=10_000
+        The total number of samples to use for creating each meta-dataset, which
+        is then split into a context set and a query set for each step of the fine-tuning loop.
+        This same context size is used during final inference as well.
+
+    finetune_split_ratio : float, default=0.2
+        The proportion of each meta-dataset to use for calculating the loss.
+        The rest is used as the context for the model.
+
+    meta_batch_size : int, default=1
+        The number of meta-datasets to process in a single optimization step.
+        Currently, this should be kept at 1.
+
+    random_state : int, default=42
+        Seed for reproducibility of data splitting and model initialization.
+
+    early_stopping : bool, default=True
+        Whether to use early stopping based on validation ROC AUC performance.
+
+    patience : int, default=3
+        Number of epochs to wait for improvement before early stopping.
+
+    min_delta : float, default=1e-4
+        Minimum change in ROC AUC to be considered as an improvement.
+
+    **kwargs : dict
+        Additional keyword arguments to pass to the underlying TabPFNClassifier,
+        such as `n_estimators`.
+    """
+
+    def __init__(
+        self,
+        device: str = "cuda",
+        epochs: int = 5,
+        learning_rate: float = 1e-5,
+        n_inference_context_samples: int = 10_000,
+        finetune_split_ratio: float = 0.2,
```

Comment:
I'd rename this to validation ratio. In my understanding this poart of the data is used for validation while finetuning, but finetuning is obviosu given the context
```suggestion
        validation_ratio: float = 0.2,
```